guppylang/compiler/expr_compiler.py

```diff
@@ -185,8 +191,12 @@ def _if_else(
         """Builds a `Conditional`, returning context managers to build the `True` and
         `False` branch.
         """
+        cond_wire = self.visit(cond)
+        cond_ty = self.builder.hugr.port_type(cond_wire.out_port())
+        if cond_ty == OpaqueBool:
+            cond_wire = self.builder.add_op(bool_to_sum(), cond_wire)
```

Comment:
I would also expect a similar change in `_new_loop`. Not sure why tests are passing?

Reply:
The break predicate in `_new_loop` always seems to take an `Either(Tuple(some type))`

Reply:
Also makes sense following the changes in #833 👍